backend/entityservice/encoding_storage.py

```diff
@@ -0,0 +1,51 @@
+import ijson
+from entityservice.serialization import deserialize_bytes, binary_format
+
+
+def convert_encodings_from_json_to_binary(f):
+    """
+    The provided file will be contain encodings and blocking information with
+    the following structure:
+
+    {
+        "clksnblocks": [
+            ["BASE64 ENCODED ENCODING 1", blockid1, blockid2, ...],
+            ["BASE64 ENCODED ENCODING 2", blockid1, ...],
+            ...
+        ]
+    }
+
+    .. Note::
+        Entities belonging to no blocks get ignored.
+
+
+    TODO:
+        Currently we do everything in memory.
+        Eventually we might want to instead stream through the input (perhaps twice).
+
+    :param f: File-like object containing `clksnblocks` json.
+    :return: a tuple comprising:
+        - dict mapping blocks to lists of bytes (each encoding in our internal encoding file format)
+        - the size of the first encoding in bytes (excluding entity ID info)
+    """
+    # Each block index contains a set of base64 encodings.
+    encodings_by_block = {}
```

Comment:
I don't think it is a good idea to store this in memory. If every encoding is in 10 different blocks, than  this structure is 10 times bigger than the already big json input `f`. I'd prefer if we would use a streaming solution that writes the encodings straight to file.